packages/api/src/utils/logs.ts

```diff
@@ -0,0 +1,310 @@
+      this.logEventsBatch = []
+      const rHost: string = batchInFlight[0].metadata.request.headers.host
+      const body = JSON.stringify(batchInFlight)
+      const request = {
+        method: 'POST',
+        headers: {
+          Authorization: `Bearer ${this.opts?.token ?? ''}`,
+          'Content-Type': 'application/json',
+          'User-Agent': `Cloudflare Worker via ${rHost}`
+        },
+        body
+      }
+      const resp = await fetch(logtailApiURL, request)
+      if (this.opts?.debug != null) {
+        console.info(
+          `[${this._date()}]`,
+          `${batchInFlight.length} logs pushed with status ${resp.status}.`
+        )
+      }
+    }
+  }
+
+  /**
+   * End instance, push logs and servers timings
+   *
+   * @param {Response} response
+   */
+  async end (response: Response) {
+    if (this._finished) {
+      throw new Error(
+        `end() has already been called on this Logging instance.
+        You must make a new instance per request.`
+      )
+    }
+    this._finished = true
+    if (this.opts?.debug != null) {
+      response.headers.set('Server-Timing', this._timersString())
+    }
+    // Automatically stop the timer of the request
+    this.timeEnd('request')
+
+    const run = async () => {
+      const dt = this._date()
+      const duration = Date.now() - this.startTs
+      const log = {
+        message: '',
+        dt,
+        level: 'info',
+        metadata: {
+          ...this.metadata,
+          response: {
+            headers: buildMetadataFromHeaders(response.headers),
+            status_code: response.status,
+            duration
+          }
+        }
+      }
+      this._add(log)
+      await this.postBatch()
+    }
+    this.ctx.waitUntil(run())
+    return response
+  }
+
+  /**
+   * Log
+   *
+   * @param {string | Error} message
+   * @param {'debug' | 'info' | 'warn' | 'error'} level
+   * @param {any} [context]
+   * @param {any} [metadata]
+   */
+  log (message: string | Error, level: LogLevels, context: any, metadata?: any) {
+    const dt = this._date()
+    let log = {
+      dt,
+      level,
+      metadata: { ...this.metadata, ...metadata },
+      ...context
+    }
+
+    // This array of errors not to send to Sentry could be configurable in the
+    // constructor if we want to keep this Logging class more generic
+    const skipForSentry = this.opts.skipForSentry
+
+    if (message instanceof Error) {
+      log = {
+        ...log,
+        stack: message.stack,
+        message: message.message
+      }
+      if (
+        this.opts.sentry != null &&
+        (skipForSentry != null) &&
+        !skipForSentry.some((cls) => message instanceof Error)
+      ) {
+        this.opts.sentry.captureException(message)
+      }
+      if (this.opts?.debug != null) {
+        console[level](`[${dt}]`, message)
+      }
+    } else {
+      log = {
+        ...log,
+        message
+      }
+      if (this.opts?.debug != null) {
```

Comment:
Ditto re. null comparison.

Reply:
Updated to `?? false` as per https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/strict-boolean-expressions.md